pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -1248,6 +1238,57 @@ func (p *Planner) ensureRKEStateSecret(controlPlane *rkev1.RKEControlPlane, full
 	}, nil
 }
 
+func getIPAddresses(secrets corecontrollers.SecretCache, machine *capi.Machine) (string, string) {
+	internalIPAddress, internalAddressProvided := machine.Annotations[InternalAddressAnnotation]
+	ipAddress, addressProvided := machine.Annotations[AddressAnnotation]
+
+	// Unmanaged machines won't have a state secret
+	if machine.Annotations["objectset.rio.cattle.io/id"] != "unmanaged-machine" && (!internalAddressProvided || !addressProvided) {
+		secret, err := secrets.Get(machine.Spec.InfrastructureRef.Namespace, name.SafeConcatName(machine.Spec.InfrastructureRef.Name, "machine", "state"))
+		if err == nil && secret != nil && len(secret.Data["extractedConfig"]) != 0 {
```

Comment:
The "contract" of caches/clients is if `err == nil` then secret != nil.  If it was nil, you'd get an IsNotFound error.  So you don't have to check for secret != nil

Reply:
Done.